httpie/core.py

```diff
@@ -93,6 +91,13 @@ def main(
                 f'Too many redirects'
                 f' (--max-redirects={parsed_args.max_redirects}).'
             )
+        except requests.ConnectionError:
+            exit_status = ExitStatus.ERROR
+            env.log_error(
+                f"Connection aborted while doing a {parsed_args.method} request to URL: {parsed_args.url}"
```

Comment:
'Single quotes please'

Reply:
Actually, one more thing: the displayed URL should be taken from the request, not from the args. It’s because there is some additional processing and the final URL may differ. For example here

```bash
$ http xxx.example.org/../foo
```

The URL ends up being `http://xxx.example.org/foo` (added protocol + normalized path), as can be seen in the current error message:

`http: error: ConnectionError: HTTPConnectionPool(host='xxx.example.org', port=80): Max retries exceeded with url: /foo (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x10a5c86d0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')) while doing a GET request to URL: http://xxx.example.org/foo`

The test would also be good to update to work with some processed URL like this one.

Reply:
Nice catch! I've updated the exception block so that the displayed URL is now of format {request.URL}